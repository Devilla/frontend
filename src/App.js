import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
