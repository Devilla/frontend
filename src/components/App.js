import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
