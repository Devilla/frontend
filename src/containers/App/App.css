@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
