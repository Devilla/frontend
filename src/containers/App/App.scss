@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
