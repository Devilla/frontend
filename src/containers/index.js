export { default as App } from './App/App';
export { default as DashboardContainer } from './DashboardContainer/DashboardContainer';
export { default as ConnectPage } from './ConnectPage/ConnectPage';
export { default as AnalyticsContainer } from './AnalyticsContainer/AnalyticsContainer';
export { default as NewCampaignContainer } from './NewCampaignContainer/NewCampaignContainer';
export { default as OauthContainer } from './OauthContainer/OauthContainer';
