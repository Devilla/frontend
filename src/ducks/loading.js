import { Map } from 'immutable';

const action = name => `zikher/modals/${name}`;
const LOAD = action('LOAD');
const LOADED = action('LOADED');

export const load = () => ({ type: LOAD });
export const loaded = () => ({ type: LOADED });

const initialState = Map({
  state: false,
});

const loading = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return state.set('state', true);

    case LOADED:
      return state.set('state', false);

    default:
      return state;
  }
};

export default loading;
