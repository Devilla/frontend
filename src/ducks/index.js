import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
