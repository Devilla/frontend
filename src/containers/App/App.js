import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
