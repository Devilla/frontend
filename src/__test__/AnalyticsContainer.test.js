import React from 'react';
import '../setupTests';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { AnalyticsContainer } from 'containers';
import { store } from 'App';

// Snapshot for Home React Component
describe('>>> AnalyticsContainer --- Snapshot',()=>{
    it('+++capturing Snapshot of AnalyticsContainer', () => {
        const renderedValue =  renderer.create(<AnalyticsContainer store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('check for in', ()=> {
  let wrapper;
  beforeEach ( () => {
       wrapper = shallow(<AnalyticsContainer  store={store}/>);
  })

  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });


  it('+++ contains website - error', () => {
    let wrapper = shallow(<AnalyticsContainer  store={store}/>);
    expect(wrapper.find('.website-error').length).toBe(0);
  });

  it('+++ contains website - analytics', () => {
    let wrapper = shallow(<AnalyticsContainer  store={store}/>);
    expect(wrapper.find('td').length).toBe(0);
  });

});
