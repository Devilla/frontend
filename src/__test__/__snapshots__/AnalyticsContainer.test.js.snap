// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> AnalyticsContainer --- Snapshot +++capturing Snapshot of AnalyticsContainer 1`] = `
<div
  className="analytics-container"
>
  <div
    className="card-box"
  >
    <h4
      className="m-t-0 header-title"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <i
          className="icon-arrow-left mr-3"
        />
      </a>
      Analytics
    </h4>
    <table
      className="table table-striped"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th
            className="text-center"
          >
            DOMAIN
          </th>
          <th
            className="text-center"
          >
            TOTAL VISITORS
          </th>
          <th
            className="text-center"
          >
            SIGNUPS
          </th>
          <th
            className="text-center"
          >
            LIVE STATS
          </th>
          <th
            className="text-center"
          >
            CONVERSION %
          </th>
          <th
            className="text-center"
          >
            Visualize
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Nothing to display
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
